### ISSUE TITLE
scipy tests failing for precision 

### ISSUE THREAD

**Issue by VictorRodriguez** [scipy/scipy#7535]
The bug was reported with similar description in : 
https://github.com/scipy/scipy/pull/4243

The AssertionError:
Not equal to tolerance rtol=1e-07, atol=1e-14
is still present

Is there any special change that needs to be done on openblas ( what i am using ) compiling flags?
 
### Reproducing code example:
```
improt scipy
print(scipy.test("full"))
```

### Error message:
```
======================================================================
FAIL: test_basic (test_signaltools.TestFiltFilt)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/scipy/signal/tests/test_signaltools.py", line 1397, in test_basic
    assert_allclose(out, arange(12), atol=1e-14)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Not equal to tolerance rtol=1e-07, atol=1e-14

(mismatch 8.33333333333%)
 x: array([ -1.634248e-12,   1.000000e+00,   2.000000e+00,   3.000000e+00,
         4.000000e+00,   5.000000e+00,   6.000000e+00,   7.000000e+00,
         8.000000e+00,   9.000000e+00,   1.000000e+01,   1.100000e+01])
 y: array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11])

======================================================================
FAIL: test_basic (test_signaltools.TestSOSFiltFilt)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/scipy/signal/tests/test_signaltools.py", line 1397, in test_basic
    assert_allclose(out, arange(12), atol=1e-14)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Not equal to tolerance rtol=1e-07, atol=1e-14

(mismatch 8.33333333333%)
 x: array([ -1.634248e-12,   1.000000e+00,   2.000000e+00,   3.000000e+00,
         4.000000e+00,   5.000000e+00,   6.000000e+00,   7.000000e+00,
         8.000000e+00,   9.000000e+00,   1.000000e+01,   1.100000e+01])
 y: array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11])

======================================================================
FAIL: test_qhull.TestUtilities.test_degenerate_barycentric_transforms
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/site-packages/scipy/spatial/tests/test_qhull.py", line 343, in test_degenerate_barycentric_transforms
    assert_(bad_count < 20, bad_count)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 92, in assert_
    raise AssertionError(smsg)
AssertionError: 20

======================================================================
FAIL: test_qhull.TestUtilities.test_more_barycentric_transforms
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/site-packages/scipy/spatial/tests/test_qhull.py", line 407, in test_more_barycentric_transforms
    unit_cube_tol=1e7*eps)
  File "/usr/lib/python2.7/site-packages/scipy/spatial/tests/test_qhull.py", line 329, in _check_barycentric_transforms
    assert_(ok.all(), "%s %s" % (err_msg, np.where(~ok)))
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 92, in assert_
    raise AssertionError(smsg)
AssertionError: ndim=4 (array([ 8071, 10512, 10513]),)

======================================================================
FAIL: test_orthogonal.test_roots_jacobi
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/site-packages/scipy/special/tests/test_orthogonal.py", line 324, in test_roots_jacobi
    vgq(rf(18.24, 27.3), ef(18.24, 27.3), wf(18.24, 27.3), -1., 1., 25)
  File "/usr/lib/python2.7/site-packages/scipy/special/tests/test_orthogonal.py", line 287, in verify_gauss_quad
    assert_allclose(vv, np.eye(N), rtol, atol)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Not equal to tolerance rtol=1e-15, atol=1e-14

(mismatch 0.32%)
 x: array([[  1.000000e+00,  -9.107374e-16,  -2.974619e-16,  -1.468511e-16,
          1.018156e-16,   3.736521e-16,   7.011539e-16,  -1.840656e-16,
         -1.269248e-15,  -1.069845e-16,   1.338407e-15,   2.493098e-16,...
 y: array([[ 1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,
         0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,...

======================================================================
FAIL: test_orthogonal.test_roots_genlaguerre
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/site-packages/scipy/special/tests/test_orthogonal.py", line 711, in test_roots_genlaguerre
    vgq(rootf(0.1), evalf(0.1), weightf(0.1), 0., np.inf, 100, atol=1e-13)
  File "/usr/lib/python2.7/site-packages/scipy/special/tests/test_orthogonal.py", line 287, in verify_gauss_quad
    assert_allclose(vv, np.eye(N), rtol, atol)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Not equal to tolerance rtol=1e-15, atol=1e-13

(mismatch 0.28%)
 x: array([[  1.000000e+00,  -9.168102e-15,  -5.951562e-15, ...,
         -9.701575e-17,  -6.021814e-17,  -2.107937e-17],
       [ -9.223791e-15,   1.000000e+00,  -6.566734e-15, ...,...
 y: array([[ 1.,  0.,  0., ...,  0.,  0.,  0.],
       [ 0.,  1.,  0., ...,  0.,  0.,  0.],
       [ 0.,  0.,  1., ...,  0.,  0.,  0.],...

----------------------------------------------------------------------
Ran 25606 tests in 394.040s

FAILED (KNOWNFAIL=128, SKIP=2053, failures=6)
<nose.result.TextTestResult run=25606 errors=0 failures=6>
```

### Scipy/Numpy/Python version information:
```
<<Output from 'import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)'>>

>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
('0.19.1', '1.13.0', sys.version_info(major=2, minor=7, micro=12, releaselevel='final', serial=0))

```

**Comment by rgommers** [comment]
@VictorRodriguez the mismatches here are really small, so probably nothing much is wrong - only test tolerances just too tight. If so, we should adjust the tests.

Can you report your OS and compiler versions?

**Comment by VictorRodriguez** [comment]
@rgommers I agree, it can be fixed with the test tolerances, this is the compiler and OS version: 

gcc (Clear Linux OS for Intel Architecture) 7.1.1 20170509

**Comment by VictorRodriguez** [comment]
Please let me know how can I adjust the tolerance on the tests , regards

**Comment by fenrus75** [comment]
docker run -it clearlinux/machine-learning

gets you a local docker (with shell into it) that has all the bits needed to repro

**Comment by rgommers** [comment]
> Please let me know how can I adjust the tolerance on the tests , regards

The first line of each test failure contains the location to modify:

      File "/usr/lib/python2.7/site-packages/scipy/signal/tests/test_signaltools.py", line 1397, in test_basic
    assert_allclose(out, arange(12), atol=1e-14)

So change ``atol`` from 1e-14 to 5e-13 and then keep increasing it slowly till the test passes.

**Comment by sethtroisi** [comment]
Fixed in #11181
